lib/rubocop/cop/rspec/describe_class.rb

```diff
@@ -27,36 +40,42 @@ class DescribeClass < Base
         MSG = 'The first argument to describe should be '\
               'the class or module being tested.'
 
-        def_node_matcher :rails_metadata?, <<-PATTERN
-          (pair
-            (sym :type)
-            (sym { :channel :controller :helper :job :mailer :model :request
-                   :routing :view :feature :system :mailbox })
-          )
-        PATTERN
-
-        def_node_matcher :example_group_with_rails_metadata?, <<~PATTERN
-          (send #rspec? :describe ... (hash <#rails_metadata? ...>))
+        def_node_matcher :example_group_with_ignored_metadata?, <<~PATTERN
+          (send #rspec? :describe ... (hash <#ignored_metadata? ...>))
         PATTERN
 
         def_node_matcher :not_a_const_described, <<~PATTERN
           (send #rspec? :describe $[!const !#string_constant?] ...)
         PATTERN
 
-        def on_top_level_group(top_level_node)
-          return if example_group_with_rails_metadata?(top_level_node.send_node)
+        def on_top_level_group(node)
+          return if example_group_with_ignored_metadata?(node.send_node)
 
-          not_a_const_described(top_level_node.send_node) do |described|
+          not_a_const_described(node.send_node) do |described|
             add_offense(described)
           end
         end
 
         private
 
+        def ignored_metadata?(node)
+          return false unless sym_pair?(node)
+
+          ignored_metadata[node.key.value.to_s]&.include? node.value.value.to_s
+        end
+
+        def sym_pair?(node)
+          node.key.sym_type? && node.value.sym_type?
+        end
```

Comment:
This can be defined as:
```ruby
        def_node_matcher :sym_pair?, <<~PATTERN
          (pair sym sym)
        PATTERN
```
if you like.

Reply:
Or even
```rb
def_node_matcher :sym_pair?, <<~PATTERN
  (pair $sym $sym)
PATTERN
```
(capturing the key and value) – in which case `#ignored_metadata?` could be reduced to
```rb
def ignored_metadata?(node)
  sym_pair?(node) do |key, value|
    ignored_metadata[key.value.to_s]&.include?(value.value.to_s)
  end
end
```

Reply:
Wondering if it needs to be `sym_pair?(node).any? |key, value|`?